fw/http_frame.c

```diff
@@ -600,16 +616,17 @@ __tfw_h2_stream_unlink(TfwH2Ctx *ctx, TfwStream *stream)
 
 	if (hmreq) {
 		hmreq->stream = NULL;
+
 		/*
-		 * If the request is linked with a stream, but not complete yet,
-		 * it must be deleted right here to avoid leakage, because in
-		 * this case it is not used anywhere yet. When request is
-		 * assembled and complete, it will be removed (due to some
-		 * processing error) in @tfw_http_req_process(), or in other
-		 * cases controlled by server connection side (after adding to
-		 * @fwd_queue): successful response sending, eviction etc.
-		 */
-		if (!test_bit(TFW_HTTP_B_FULLY_PARSED, hmreq->flags))
+		* If the request is linked with a stream, but not complete yet,
+		* it must be deleted right here to avoid leakage, because in
+		* this case it is not used anywhere yet. When request is
+		* assembled and complete, it will be removed (due to some
+		* processing error) in @tfw_http_req_process(), or in other
+		* cases controlled by server connection side (after adding to
+		* @fwd_queue): successful response sending, eviction etc.
+		*/
+		if (!test_bit(TFW_HTTP_B_FULLY_PARSED, hmreq->flags) && !hmreq->resp)
 			tfw_http_conn_msg_free(hmreq);
```

Comment:
Does `!hmreq->resp` implies websockets? If so, then we should make an assertion that this is a websocket request, otherwise this isn't clear. If not, then where the request with empty `resp` is freed now for regular HTTP/2 messages?